pypesto/petab/importer.py

```diff
@@ -78,6 +80,38 @@ def from_yaml(yaml_config: Union[dict, str],
             output_folder=output_folder,
             model_name=model_name)
 
+    def check_and_log_gradients(self, RTOL: float = 1e-2, ATOL: float = 1e-3):
+        """Check and log if gradients are compatible with gradient-based optimizers
+        Parameters
+        ----------
+        RTOL: relative error tolerance (default 1e-2)
+        ATOL: absolute error tolerance (default 1e-3)
+        """
+        par = np.asarray(self.petab_problem.x_nominal_scaled)
```

Comment:
Could be by default but also accept a user-supplied vector.

Reply:
Made as default and allow user defined tolerances.